src/include/compat.h

```diff
@@ -13,10 +13,57 @@
 #define CEPH_COMPAT_H
 
 #if defined(__FreeBSD__)
+
+/* Make sure that ENODATA is defined in the correct way */
+#ifndef ENODATA
 #define	ENODATA	ENOATTR
+#else
+#if (ENODATA == 9919)
+// #warning ENODATA already defined to be 9919, redefining to fix
```

Comment:
@tchaikov 
I think you are refering to the text in the discussion, not the actual commit?
The current commit is all done during initial parsing by the pre-processor, nothing special there.
However: The amount of missery that will occur when ENODATA is actually the wrong value is rather serious. And as such it would IMHO a compile time check when the definition is actually used.
But that does require a more serious macro to be writting, which needs to be expanded upon reference, not upon inclusion.
But that is left for (far) future work, perhaps it is best to ignore that.